### PR TITLE
Fixing bugs found in v3-beta during testing.

### PR DESCRIPTION
**Pull request by Stormsys**

### CODE REVIEW

File: src/ZendeskApi.Client.Abstractions/Resources/ISearchResource.cs [hunk 1]
```diff
@@ -9,5 +9,7 @@ namespace ZendeskApi.Client.Resources
     public interface ISearchResource
     {
         Task<IPagination<ISearchResult>> SearchAsync(Action<IZendeskQuery> builder, PagerParameters pager = null);
+
+        Task<IPagination<T>> SearchAsync<T>(Action<IZendeskQuery> builder, PagerParameters pager = null) where T : ISearchResult;
```
Comment (Jetski5822) [inline src/ZendeskApi.Client.Abstractions/Resources/ISearchResource.cs]: LOL!!!! I missed that!? ARGH

File: src/ZendeskApi.Client/Queries/ZendeskQueryTicketExtensions.cs [hunk 2]
```diff
@@ -24,17 +24,22 @@ public static IZendeskQuery WithTicketStatus(this IZendeskQuery query, TicketSta
 
         public static IZendeskQuery WithCreatedDate(this IZendeskQuery query, DateTime dateTime, FilterOperator op)
         {
-            return query.WithFilter("created", $"{dateTime.Year}-{dateTime.Month.ToString("d2")}-{dateTime.Day.ToString("d2")}", op);
+            return query.WithFilter("created", $"{dateTime:yyyy-MM-dd}", op);
```
Comment (Jetski5822) [inline src/ZendeskApi.Client/Queries/ZendeskQueryTicketExtensions.cs]: much nicer

**Comment by Jetski5822** [comment]
LGTM

File: src/ZendeskApi.Client.Abstractions/Models/User.cs [hunk 3]
```diff
@@ -39,7 +39,7 @@ public class User : ISearchResult
         public string Locale { get; set; }
 
         [JsonProperty("locale_id")]
-        public long? LocalId { get; set; }
+        public int? LocalId { get; set; }
```
Comment (hjaltethor) [inline src/ZendeskApi.Client.Abstractions/Models/User.cs]: Int? Add -e to name

**Pull request merged**